### PR TITLE
Fix some uninitialized pointer usage in Driver

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: gejin <ge.jin@intel.com>

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -7708,7 +7708,8 @@ void SYCLPostLink::ConstructJob(Compilation &C, const JobAction &JA,
     addArgs(CmdArgs, TCArgs, {"-symbols"});
   }
   // specialization constants processing is mandatory
-  if (llvm::dyn_cast<SYCLPostLinkJobAction>(&JA)->getRTSetsSpecConstants())
+  if (llvm::dyn_cast<SYCLPostLinkJobAction>(&JA) &&
+      llvm::dyn_cast<SYCLPostLinkJobAction>(&JA)->getRTSetsSpecConstants())
```
Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:7712]: Optional:
```suggestion
  auto *SYCLPostLink = llvm::dyn_cast<SYCLPostLinkJobAction>(&JA);
  if (SYCLPostLink && SYCLPostLink->getRTSetsSpecConstants());
```

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 2]
```diff
@@ -672,9 +672,6 @@ RTDeviceBinaryImage &ProgramManager::getDeviceImage(OSModuleHandle M,
   return *Img;
 }
 
-// TODO: getDeviceLibPrograms should also support Windows but
-// current implementation doesn't work on Windows when multiple
-// device libraries exist and this problem should be fixed.
```
Comment (AGindinson) [inline sycl/source/detail/program_manager/program_manager.cpp:677]: Could you please remove the unrelated changes?

**Comment by AGindinson** [comment]
A `[SYCL]` tag would also be desired for the commit message. Thanks in advance!

File: clang/lib/Driver/ToolChains/Clang.cpp (hunk 1 above)
Comment (jinge90) [reply clang/lib/Driver/ToolChains/Clang.cpp]: Already done

**Comment by jinge90** [comment]
> A `[SYCL]` tag would also be desired for the commit message. Thanks in advance!

The tag is added. Thank you for kind remind.

File: sycl/source/detail/program_manager/program_manager.cpp (hunk 2 above)
Comment (jinge90) [reply sycl/source/detail/program_manager/program_manager.cpp:677]: Several months ago, we found device library infrastructure can't work on Windows for multiple device libraries. We thought it was sycl runtime's bug so added the "TODO" here but it turned out the root cause was in underlying runtime and they have fixed this issue a bout 1 month ago. So we can directly remove the comments here without any code change.
Thank you very much.

Comment (jinge90) [reply sycl/source/detail/program_manager/program_manager.cpp]: Already removed the unrelated change.
